src/hooks/search/meta.ts

```diff
@@ -0,0 +1,76 @@
+import getConfig from 'next/config';
+import { useRouter } from 'next/router';
+import { useCallback, useState } from 'react';
+import { getCanonicalUrlWithParams, getMetaDescr } from '../../utils/seo';
+import { replaceHistoryState } from './utils';
+
+export const useSearchMeta = (
```

Comment:
this hook is pretty self explanatory but I was seeing a lot of duplicated page-meta-related code and it tidies things up greatly to have that stuff centralized here